signin.js

```diff
@@ -1,90 +1,55 @@
-// 테스트 계정 설정
-const testCase = {
-  email: 'test@codeit.com',
-  password: 'codeit101',
-}
-
-// 이메일 형식 확인하는 정규화
-const emailForm = /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i
+import { accountInfo, emailForm, passwordFrom } from "./javascript/account.js"
+import { inputFunction } from "./javascript/signFunction.js"
+import { passwordToggle } from "./javascript/passwordToggle.js"
 
 // 
 const inputEmail = document.querySelector('.email-input')
 const inputPassword = document.querySelector('.password-input')
-const loginButton = document.querySelector('.sign-btn')
-const togglePassword = document.querySelector('.eye-btn')
-
-// email foucus, blur 함수
-function emailFocus(e) {
-  inputEmail.classList.remove('input-error')
-  e.target.parentElement.lastElementChild.textContent = ''
-}
+const signButton = document.querySelector('.sign-btn')
 
-function emailBlur(e) {
-  if (e.target.value === '') {
-    e.target.classList.add('input-error')
-    e.target.parentElement.lastElementChild.textContent = '이메일을 입력해 주세요.'
-  }
-  else if (!e.target.value.match(emailForm)) {
-    e.target.classList.add('input-error')
-    e.target.parentElement.lastElementChild.textContent = '올바른 이메일 주소가 아닙니다.'
-  }
-}
-
-inputEmail.addEventListener('focus', emailFocus)
-inputEmail.addEventListener('blur', emailBlur)
+// 로그인 버튼, 경우에 따른 케이스 적용
+const submitButton = (e) => {
```

Comment:
해당 함수 가독성이 떨어지네요. 밑의 input error case가 더 정리되면 좋겠습니다. 조건들을 변수로 빼던지 아니면 emailValue, passwordValue를 받아서 어떤 에러인지 반환하는 함수를 만들면 좋겠네요.

```
// 조건문을 단순화한 경우
const isEmailInvalid = !matchingAccountEmail?.email 
const isPasswordInvalid = !matchingAccountPassword?.password

if(isEmailInvalid && isPasswordInvalid) return ...

// email, password input의 값을 받아 어떤 에러인지 반환하는 함수
const getErrorType = (email, password) => {
  if(이메일 비밀번호 모두 불일치) return 'bothInvalid';
  if(이메일 중복) return 'duplicatedEmail';
  ...
}
```

Reply:
함수 이름에 button이 들어갈필요도 없어보입니다.
onSubmit 같은 이름이 더 좋을 것 같습니다.

```suggestion
const onSubmit = (e) => {
```

Reply:
네. 좀 더 가독성 부분을 고민해서 코드를 수정해보겠습니다!!